### ISSUE TITLE
Support for Function Calls

### ISSUE THREAD

**Issue by keyofdb** [TomFrost/Jexl#25]
Does this library support function calls?

For example, given the following context
{
  number1: 3,
  number2: 5,
  number3: 4
}

Can i use something like the following?
min(number1, number2, number3)

**Comment by keyofdb** [comment]
Sorry meant 
min(number1, number2, number3)

**Comment by johannesschobel** [comment]
that would be so cool.. 

@TomFrost : maybe this feature can be added to `v2.0`? That would greatly enhance this awesome library!

**Comment by TomFrost** [comment]
It's been a long time coming, but functions support is working in the [functions branch](https://github.com/TomFrost/Jexl/tree/functions). No docs there yet, but it works exactly as you might expect:

```javascript
jexl.addFunction('hello', (arg) => `Hello, ${arg}!`)
jexl.evalSync('hello("world")') // "Hello, world!"
```

Getting this documented and teed up for the next release!

**Comment by austinkelleher** [comment]
This is going to be fantastic! Thanks for your work @TomFrost!

**Comment by shrugal** [comment]
Adding functions to the context would be nice, e.g.:

```javascript
jexl.eval("a.b(5).c", { a: { b: v => ({ c: v }) } })
```

**Comment by austinkelleher** [comment]
Any update on this @TomFrost? Is it safe to use the functions branch?

**Comment by TomFrost** [comment]
Released in 2.3.0!
